app/state_machines/person_escort_record_state_machine.rb

```diff
@@ -0,0 +1,22 @@
+class PersonEscortRecordStateMachine < FiniteMachine::Definition
+  initial :unstarted
+
+  event :complete, %i[unstarted in_progress completed] => :completed
```

Comment:
we have to allow completed to be run on completed if a response is updated, and the same for in_progress state as well to allow further update to responses

Reply:
Even so, I don't think you need to map that state transition. Does calling `complete` when the state is already completed raise an exception or is it ignored instead? If it raises an exception then I'd agree you need this.

Reply:
Maybe this is the wrong approach to take with the state machine. It might be cleaner/preferable to add guard clauses on the transition and add a simple `update` event in the state machine. You can always call that from the controller, and within the state machine check whether progress is complete or not and make the correct transition. Something like this:

```ruby
event :update, %i[unstarted in_progress] => :completed, if: -> { target.progress == PERSON_ESCORT_RECORD_IN_PROGRESS }
event :update, %i[completed] => :in_progress, if -> { target.progress == PERSON_ESCORT_RECORD_IN_PROGRESS }
```

Then in the controller

```ruby
def update_status!
  self.progress = set_progress(sections_to_responded)
  state_machine.update!
  save!
end
```

Something to think about, but I believe this is a more idiomatic approach for the state machine.